lfsapi/client.go

```diff
@@ -122,10 +124,33 @@ func (c *Client) doWithRedirects(cli *http.Client, req *http.Request, via []*htt
 		return nil, err
 	}
 
-	res, err := cli.Do(req)
+	var retries int
+	if n, ok := Retries(req); ok {
+		retries = n
+	} else {
+		retries = defaultRequestRetries
+	}
+
+	retries = tools.MaxInt(0, retries)
+
+	var res *http.Response
+
+	for i := 0; i < retries; i++ {
```

Comment:
`retries` might a bit misleading here. The first "retry" is the actual request, no? Therefore `retries` should always be greater `0`, no?

Reply:
Good point -- I like the name retries, but I think the current interpretation of it is a little bit off. Let's treat 'Retries' as the number of _additional_ requests to make for a failed request/response cycle, and instead calculate `retries` (lowercase `r`) as:

```go
retries := tools.MaxInt(0, Retries(req)) + 1
```